src/functions/dialog/index.ts

```diff
@@ -5,38 +5,68 @@
 import type { Component } from 'vue'
 import { createApp, toRaw } from 'vue'
 
-interface DialogProps {
-	[index: string]: unknown
-	container?: string
+type SpawnDialogOptions = {
+	/**
+	 * Container to mount the dialog to
+	 * @default document.body
+	 */
+	container?: Element | string
 }
 
+export function spawnDialog(
+	dialog: Component,
+	props?: object,
+	onClose?: (...rest: unknown[]) => void,
+): void
+
+export function spawnDialog(
+	dialog: Component,
+	props?: object,
+	options?: SpawnDialogOptions,
+	onClose?: (...rest: unknown[]) => void,
+): void
+
 /**
- * Helper to spawn a Vue dialog without having to mount it from a component
+ * Spawn a single-use Vue dialog instance to get the result when it is closed
  *
- * @param dialog The dialog component to spawn
- * @param props Properties to pass to the dialog
- * @param props.container Optionally pass a query selector for the dialog container element
- * @param onClose Callback when the dialog is closed
+ * @param dialog - Dialog component to spawn
+ * @param props - Props to pass to the dialog instance
+ * @param optionsOrOnClose - Spawning options or a callback when the dialog is closed
+ * @param onClose - Callback when the dialog is closed
  */
 export function spawnDialog(
 	dialog: Component,
-	props?: DialogProps,
+	props: object = {},
+	optionsOrOnClose: SpawnDialogOptions | ((...rest: unknown[]) => void) = {},
 	onClose: (...rest: unknown[]) => void = () => {},
 ): void {
-	const el = document.createElement('div')
-	const container: HTMLElement = typeof props?.container === 'string'
-		? (document.querySelector(props.container) || document.body)
-		: document.body
-	container.appendChild(el)
+	if (typeof optionsOrOnClose === 'function') {
+		onClose = optionsOrOnClose
+		optionsOrOnClose = {}
+	}
+
+	let { container } = optionsOrOnClose
+
+	// For backwards compatibility try to use container from props
+	if ('container' in props && typeof props.container === 'string') {
```

Comment:
```suggestion
	if (container&& 'container' in props && typeof props.container === 'string') {
```

Do not override if already in opts

Reply:
Did
```
if ('container' in props && typeof props.container === 'string') {
	container ??= props.container
}
```
instead.

(Seems slightly more readable to me to not think about brackets in `!container && 'container' in props`)

Reply:
`!container && 'container' in props` vs `!container && ('container' in props)`